pkg/server/commands/listusers/list_users.go

```diff
@@ -12,3 +14,87 @@ type listUsersRequest interface {
 	GetUserFilters() []*openfgav1.ListUsersFilter
 	GetContextualTuples() *openfgav1.ContextualTupleKeys
 }
+
+type internalListUsersRequest struct {
```

Comment:
**Thinking out loud:** This mostly-redundant struct seems clunky but I'm wondering maybe if that's because we haven't wired this up to the server yet. Once we do that, we'll have abstracted the server layer away from the underlying algorithm, at which point we can rename this to something more elegant. 

WDYT?

Reply:
i don't think it's clunky, i agree it's not pretty, but we need to keep track of internal state somewhere. Since we cannot extend protobuf objects, i created my own. I could also do

```go
type internalListUsersRequest struct {
     openfgav1.ListUsersRequest

     visitedUsersetsMap map[string]struct{}

}
```

Reply:
If this were implemented as an RPC, what would we do? Assume the ListUsersRequest is an internal RPC definition and would otherwise be dispatchable over any interface.

In this case, we'd "extend" the base request. Namely,

```
type DispatchListUsersRequest struct {
    openfgav1.ListUsersRequest
    ... // other metadata applicable to the internal dispatch
}
```

Reply:
I ended up doing ☝️